helix-core/src/main/java/org/apache/helix/util/HelixUtil.java

```diff
@@ -382,13 +382,16 @@ public static Map<String, Map<String, String>> getIdealAssignmentForFullAuto(
       allNodes.add(instanceConfig.getInstanceName());
       instanceConfigMap.put(instanceConfig.getInstanceName(), instanceConfig);
     }
-    ResourceControllerDataProvider cache = new ResourceControllerDataProvider();
-    cache.setClusterConfig(clusterConfig);
-    cache.setInstanceConfigMap(instanceConfigMap);
 
+    // TODO: Consider full cache refresh to prevent needing to manually set necessary fields
```

Comment:
I guess core does not impact, right? Only rest?

Reply:
Merge this change now for the 1.4.1 release. Please think of impact.

Reply:
This method `getIdealAssignmentForFullAuto` is only called by `ResourceAssignmentOptimizerAccessor` in helix-rest module (the partitionAssignment API). It seems the only impact was partitionAssignment API calls for FULL_AUTO resources that are not on WAGED. Core should not be affected from what I've seen. 

The equivalent ideal state calculation method used by partitionAssignment API for WAGED resources `getAssignmentForWagedFullAutoImpl` creates and fully refreshes the ResourceControllerDataProvider.